Testscripts/Linux/validate-da.sh

```diff
@@ -220,20 +237,22 @@ function enable_disable_da(){
         fail_test "$da_log_dir/service.log.2 exist."
     fi
     
-    if [ ! -c "/dev/msda" ]; then 
-        fail_test "/dev/msda does not exist."
-    fi
-    
     # Check for "driver setup status=0" and "starting the dependency agent" in service.log.1 and service.log respectively
     
-    if ! driver_status=$(sed -n '/Driver setup status=//s/^[^=]*//p' $da_log_dir/service.log.1); then
-        fail_test "sed command failed while reading service.log.1 file"
+    if ! driver_status=$(sed -n '/Driver setup status=/s/^[^=]*//p' "$da_log_dir/service.log.1"); then
+        fail_test "sed command failed while reading $da_log_dir/service.log.1 file"
     fi
 
     case "$driver_status" in
         =0)
             # expected result
             ;;
+        =36)
+            # Unsupported kernel case
```

Comment:
Please use the pattern `=3[67]` for this case because we want to catch if either module fails to load.

The only "expected" reason for the module load to fail is if the kernel doesn't have Spectre mitigation. Please add a check that if the file `/sys/devices/system/cpu/vulnerabilities/spectre_v2` exists and the contents start with "Mitigation:", then getting one of these two statuses is test failure. Otherwise, we skip some of the assertions.

Also, if the test fails above, please use `tail` to capture the last 20 lines of the service log into the test error log.

Reply:
Done.